app/services/course/statistics/assessments_score_summary_download_service.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+require 'csv'
+class Course::Statistics::AssessmentsScoreSummaryDownloadService
+  include ApplicationFormattersHelper
+
+  class << self
+    def download(course, assessment_ids)
+      service = new(course, assessment_ids)
+      ActsAsTenant.without_tenant do
+        service.generate_csv_report
+      end
+    end
+  end
+
+  def generate_csv_report
+    assessment_file_name = "#{Pathname.normalize_filename(@course.title)}_assessment_score_summary.csv"
+    assessment_score_summary_file_path = File.join(@base_dir, assessment_file_name)
+
+    load_total_grades
+    CSV.open(assessment_score_summary_file_path, 'w') do |csv|
+      download_score_summary(csv)
+    end
+
+    assessment_score_summary_file_path
+  end
+
+  private
+
+  def initialize(course, assessment_ids)
+    @course = course
+    @assessment_ids = assessment_ids
+    @base_dir = Dir.mktmpdir('assessment-score-summary-')
+  end
+
+  def load_total_grades
+    @assessment_hash = Course::Assessment.where(id: @assessment_ids).to_h do |assessment|
+      [assessment.id, assessment]
+    end
+
+    @assessments = @assessment_ids.map do |assessment_id|
+      @assessment_hash[assessment_id.to_i]
+    end
+
+    @submissions = Course::Assessment::Submission.where(assessment_id: @assessment_ids).
```

Comment:
Nice! Single query for all submissions.

Help me confirm that if I give assessment ids from other courses, it won't be able to query/download the data.

Reply:
good catch! Even though this is inherently impossible for assessment ids from other courses to be included here (from FE, it's very specific that the button is owned by the course that contains it), but it might be possible for user to directly create the API request to call this, and set the `assessment_ids` randomly (even the one that does not exist at all)

I have modified it